blog/llm-finetuning/dataset_utils.py

```diff
@@ -0,0 +1,68 @@
+import datasets
+
+
+def add_length_column(dataset):
+    df = dataset.to_pandas()
+    df["total_length"] = 0
+    for column_name in ["instruction", "input", "response"]:
+        num_words = df[column_name].astype(str).str.split().apply(len)
+        df["total_length"] += num_words
+
+    return df
+
+
+def filter_by_total_length(df, difficulty):
+    if difficulty == "easy":
+        return df[df["total_length"].between(10, 100)]
+    elif difficulty == "medium":
+        return df[df["total_length"].between(101, 200)]
+    elif difficulty == "hard":
+        return df[df["total_length"].between(200, 800)]
+
```

Comment:
nit: how about we take up to 10K samples for each dataset (e.g., `df[df["total_length"].between(10, 100)].iloc[:10000]`)?